### DISCUSSION TITLE
Performance - 1M Emails in 20 minutes

### DISCUSSION THREAD

**Discussion by eranbetzalel** [postalserver/postal#2220] [Help with using Postal]
We're looking to move from Mailgun to a private mailing server (multiple domains / IPs, mail retention logs, webhooks, etc...).

Initially I was hooked on the idea of using Postfix + Haraka, as I'm a NodeJS developer and will possibly need extra customization.

Having said that, using Postal seems even easier, as all the needed features are already implemented.

However, I did not find a lot of information regarding **Postal performance**, I know it's scalable, but I'm not sure its up to the required task at hand.

1. My performance requirement is at least 1M Emails in 20 minutes, can Postal do it?
2. If so, can you roughly estimate the number of instances and hardware I'll need for this task?

Thanks ahead for sharing your thoughts.

**Comment by willpower232** [comment]
With the right number of workers and enough resources, Postal should be able to do it. You might need to configure your database to handle lots of traffic, have a look at #324.

I wouldn't be able to estimate the number of workers you'll need, how long is a piece of string etc, but you'll definitely need to scale them out into double digits.
